kubernetes/resource_kubernetes_pod_test.go

```diff
@@ -533,6 +534,95 @@ func TestAccKubernetesPod_with_cfg_map_volume_mount(t *testing.T) {
 	})
 }
 
+func TestExpandThenFlatten_projected_volume(t *testing.T) {
```

Comment:
I added this test to double check that I was doing the expand/flatten properly, while debugging the `ExpectNonEmptyPlan: false|true` conundrum. I haven't seen other similar tests, so I'm not sure if it's desirable or if you'd rather I remove it.